modules/opensearch/includes/opensearch.search.inc

```diff
@@ -0,0 +1,33 @@
+<?php
+/**
+ * @file
+ * Implements handling of loans through OpenRuth.
+ */
+
+/**
+ * Get a list of material types from the Well.
+ */
+function opensearch_search_material_types() {
+  // Get a list of types by executing a null query and look at the facets
+  // result.
+  $options = [
+    'facets' => ['facet.type'],
+    'numFacets' => 99,
+    'reply_only' => TRUE,
+    'sort' => 'random',
+  ];
+  module_load_include('client.inc', 'ting');
+  $result = ting_do_search("*", 0, 0, $options);
```

Comment:
This warrants other functions to be moved to the `opensearch` module as well. I see that may be out of scope for this PR, but adding a `TODO` to remember this might be helpful.